pandas/_testing.py

```diff
@@ -1358,6 +1358,55 @@ def assert_series_equal(
     """
     __tracebackhide__ = True
 
+    _assert_series_values_equal(
+        left,
+        right,
+        check_dtype=check_dtype,
+        check_series_type=check_series_type,
+        check_less_precise=check_less_precise,
+        check_names=check_names,
+        check_exact=check_exact,
+        check_datetimelike_compat=check_datetimelike_compat,
+        check_categorical=check_categorical,
+        check_category_order=check_category_order,
+        check_freq=check_freq,
+        check_flags=check_flags,
+        rtol=rtol,
+        atol=atol,
+        obj=obj,
+    )
+
+    assert_index_equal(
+        left.index,
+        right.index,
+        exact=check_index_type,
+        check_names=check_names,
+        check_exact=check_exact,
+        check_categorical=check_categorical,
+        rtol=rtol,
+        atol=atol,
+        obj=f"{obj}.index",
+    )
+
+
+def _assert_series_values_equal(
+    left,
```

Comment:
yeah let's revert this and use the check_index=False

Reply:
Reverted to use of ``check_index``.

Reply:
But personally I prefer the way involving splitting of ``assert_series_equal`` into to functions.
Indeed, this function does two major things: check index equivalence and check values equivalence, so it is reasonable to split it like that.
I see that there are multiple parameters passed to ``_assert_series_values_equal``, which make this solution look not very elegant.
But there is a benefit that we do not touch public API, by not adding new parameter ``check_index``.
Please suggest which solution you prefer (@jbrockmendel suggested that we do not introduce new kwarg).

Reply:
the new kwarg isnt the worst thing in the world.  lets make it keyword-only to start the process of #38222

Reply:
ok i think this is fine (reason I don't mind `check_index` is that we have this elsewhere publicly). ok let's make this kwarg only fo r the new paramater, otherwise lgtm.

Reply:
I make this kwarg keyword-only.